ui/components/app/confirmations/alerts/alert-modal/alert-modal.tsx

```diff
@@ -0,0 +1,242 @@
+        icon: IconColor.warningDefault,
+      };
+    case Severity.Danger:
+      return {
+        background: BackgroundColor.errorMuted,
+        icon: IconColor.errorDefault,
+      };
+    default:
+      return {
+        background: BackgroundColor.infoMuted,
+        icon: IconColor.infoDefault,
+      };
+  }
+}
+
+function Header({ selectedAlert }: { selectedAlert: Alert }) {
+  const t = useI18nContext();
+  const severityStyle = getSeverityStyle(selectedAlert.severity);
+  return (
+    <>
+      <Box
+        gap={3}
+        display={Display.Block}
+        alignItems={AlignItems.center}
+        textAlign={TextAlign.Center}
+      >
+        <Icon
+          name={
+            selectedAlert.severity === Severity.Info
+              ? IconName.Info
+              : IconName.Danger
+          }
+          size={IconSize.Xl}
+          color={severityStyle.icon}
+        />
+        <Text
+          variant={TextVariant.headingSm}
+          color={TextColor.inherit}
+          marginTop={3}
+        >
+          {selectedAlert.reason ?? t('alert')}
+        </Text>
+      </Box>
+    </>
+  );
+}
+
+function AlertDetails({ selectedAlert }: { selectedAlert: Alert }) {
+  const t = useI18nContext();
+  const severityStyle = getSeverityStyle(selectedAlert.severity);
+  return (
+    <Box
+      key={selectedAlert.key}
+      display={Display.InlineBlock}
+      padding={2}
+      width={BlockSize.Full}
+      backgroundColor={severityStyle.background}
+      gap={2}
+      borderRadius={BorderRadius.SM}
+    >
+      <Text variant={TextVariant.bodySm}>{selectedAlert.message}</Text>
+      {selectedAlert.alertDetails?.length ? (
+        <Text variant={TextVariant.bodySmBold} marginTop={1}>
+          {t('alertModalDetails')}
+        </Text>
+      ) : null}
+
+      <Box as="ul" className={'alert-modal__alert-details'} paddingLeft={6}>
+        {selectedAlert.alertDetails?.map((detail, index) => (
+          <Box as="li" key={`${selectedAlert.key}-detail-${index}`}>
+            <Text variant={TextVariant.bodySm}>{detail}</Text>
+          </Box>
+        ))}
+      </Box>
+    </Box>
+  );
+}
+
+function AcknowledgeCheckbox({
+  selectedAlert,
+  onHandleAcknowledgeClick,
+  isConfirmed,
+}: {
+  selectedAlert: Alert;
+  onHandleAcknowledgeClick: (
+    selectedAlertKey: string,
+    isConfirmed: boolean,
+  ) => void;
+  isConfirmed: boolean;
+}) {
+  const t = useI18nContext();
+  const severityStyle = getSeverityStyle(selectedAlert.severity);
+  return (
+    <Box
+      display={Display.Flex}
+      padding={3}
+      width={BlockSize.Full}
+      gap={3}
+      backgroundColor={severityStyle.background}
+      marginTop={4}
+      borderRadius={BorderRadius.LG}
+    >
+      <Checkbox
+        label={t('alertModalAcknowledge')}
+        data-testid="alert-modal-acknowledge-checkbox"
+        isChecked={isConfirmed}
+        onClick={() => onHandleAcknowledgeClick(selectedAlert.key, isConfirmed)}
```

Comment:
I believe this should be `onChange` as it's an input element

```suggestion
        onChange={() => onHandleAcknowledgeClick(selectedAlert.key, isConfirmed)}
```